repo/blob/azure/azure_pit.go

```diff
@@ -0,0 +1,199 @@
+
+			versionTime, err := time.Parse(time.RFC3339Nano, vs[i].Version)
+			if err != nil {
+				return nil
+			}
+
+			if versionTime.After(versionLimit) {
+				return vs[:i]
+			}
+		}
+	}
+
+	return vs
+}
+
+// listBlobVersions returns a list of blob versions but the blob is deleted, it returns Azure's delete marker version but excludes
+// the Kopia delete marker version that is used to get around immutability protections.
+func (az *azPointInTimeStorage) listBlobVersions(ctx context.Context, prefix blob.ID, callback func(vm versionMetadata) error) error {
+	prefixStr := az.getObjectNameString(prefix)
+
+	pager := az.service.NewListBlobsFlatPager(az.container, &azblob.ListBlobsFlatOptions{
+		Prefix: &prefixStr,
+		Include: azblob.ListBlobsInclude{
+			Metadata:            true,
+			DeletedWithVersions: true, // this shows DeleteMarkers aka blobs with HasVersionsOnly set to true
+			Versions:            true,
+		},
+	})
+
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return translateError(err)
+		}
+
+		for _, it := range page.Segment.BlobItems {
+			vm := az.getVersionedBlobMeta(it)
+
+			if err := callback(vm); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (az *azPointInTimeStorage) getVersionedMetadata(ctx context.Context, blobID blob.ID) (versionMetadata, error) {
+	var vml []versionMetadata
+
+	if err := az.getBlobVersions(ctx, blobID, func(vm versionMetadata) error {
+		if !vm.Timestamp.After(az.pointInTime) {
+			vml = append(vml, vm)
+		}
+
+		return nil
+	}); err != nil {
+		return versionMetadata{}, errors.Wrapf(err, "could not get version metadata for blob %s", blobID)
+	}
+
+	if v, found := newestAtUnlessDeleted(vml, az.pointInTime); found {
+		return v, nil
+	}
+
+	return versionMetadata{}, blob.ErrBlobNotFound
+}
+
+// isAzureDeleteMarker checks for Azure created delete markers.
+func (az *azPointInTimeStorage) isAzureDeleteMarker(it *azblobmodels.BlobItem) bool {
+	var isDeleteMarker bool
+	// HasVersionsOnly - Indicates that this root blob has been deleted
+	if it.HasVersionsOnly != nil {
+		isDeleteMarker = *it.HasVersionsOnly
+	}
+
+	return isDeleteMarker
+}
+
+// maybePointInTimeStore wraps s with a point-in-time store when s is versioned
+// and a point-in-time value is specified. Otherwise, s is returned.
+func maybePointInTimeStore(ctx context.Context, s *azStorage, pointInTime *time.Time) (blob.Storage, error) {
+	if pit := s.Options.PointInTime; pit == nil || pit.IsZero() {
+		return s, nil
+	}
+
+	// IsImmutableStorageWithVersioning is needed for PutBlob with RetentionPeriod being set.
+	props, err := s.service.ServiceClient().NewContainerClient(s.container).GetProperties(ctx, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get determine if container '%s' supports versioning", s.container)
+	}
+
+	if props.IsImmutableStorageWithVersioningEnabled == nil || !*props.IsImmutableStorageWithVersioningEnabled {
```

Comment:
Can be an `IsImmutableStorageWithVersioningEnabled` check to be used in conjunction with immutability protection for now~ to check for Versioning only it seems would need either access to the related StorageAccount or to query a random blob and look for a VersionID

Reply:
Switched to versioning. If someone deleted `KopiaRepositoryBlobID` then it stops working though without user intervention to undelete it

Reply:
Potentially we could keep `IsImmutableStorageWithVersioningEnabled` as a fallback in case `KopiaRepositoryBlobID` is deleted but they have `IsImmutableStorageWithVersioningEnabled`, wdyt?